--- a/vllm/model_executor/layers/quantization/utils/ocp_mx_utils.py
+++ b/vllm/model_executor/layers/quantization/utils/ocp_mx_utils.py
@@ -1,6 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 # SPDX-FileCopyrightText: Copyright contributors to the vLLM project
 from enum import Enum
+from typing import Union
 
 from vllm.logger import init_logger
 
@@ -20,44 +21,28 @@
 
 
 class OCP_MX_Scheme(str, Enum):
-    w_mxfp4 = "w_mxfp4"
     w_mxfp4_a_mxfp4 = "w_mxfp4_a_mxfp4"
     w_mxfp4_a_mxfp6_e3m2 = "w_mxfp4_a_mxfp6_e3m2"
     w_mxfp4_a_mxfp6_e2m3 = "w_mxfp4_a_mxfp6_e2m3"
-    w_mxfp4_a_fp8 = "w_mxfp4_a_fp8"
-    w_mxfp6_e3m2 = "w_mxfp6_e3m2"
     w_mxfp6_e3m2_a_mxfp6_e3m2 = "w_mxfp6_e3m2_a_mxfp6_e3m2"
-    w_mxfp6_e3m2_a_fp8 = "w_mxfp6_e3m2_a_fp8"
-    w_mxfp6_e2m3 = "w_mxfp6_e2m3"
     w_mxfp6_e2m3_a_mxfp6_e2m3 = "w_mxfp6_e2m3_a_mxfp6_e2m3"
-    w_mxfp6_e2m3_a_fp8 = "w_mxfp6_e2m3_a_fp8"
 
     @classmethod
-    def from_quant_dtype(cls, input_dtype: str | None, weight_dtype: str | None):
-        if input_dtype not in OCP_MX_DTYPES and weight_dtype not in OCP_MX_DTYPES:
+    def from_quant_dtype(
+        cls, input_dtype: Union[str, None], weight_dtype: Union[str, None]
+    ):
+        if input_dtype not in OCP_MX_DTYPES or weight_dtype not in OCP_MX_DTYPES:
             return None
-        elif input_dtype is None and weight_dtype == "mxfp4":
-            return cls.w_mxfp4
-        elif input_dtype is None and weight_dtype == "mxfp6_e3m2":
-            return cls.w_mxfp6_e3m2
-        elif input_dtype is None and weight_dtype == "mxfp6_e2m3":
-            return cls.w_mxfp6_e2m3
         elif input_dtype == "mxfp4" and weight_dtype == "mxfp4":
             return cls.w_mxfp4_a_mxfp4
         elif input_dtype == "mxfp6_e3m2" and weight_dtype == "mxfp4":
             return cls.w_mxfp4_a_mxfp6_e3m2
         elif input_dtype == "mxfp6_e2m3" and weight_dtype == "mxfp4":
             return cls.w_mxfp4_a_mxfp6_e2m3
-        elif input_dtype == "fp8" and weight_dtype == "mxfp4":
-            return cls.w_mxfp4_a_fp8
         elif input_dtype == "mxfp6_e3m2" and weight_dtype == "mxfp6_e3m2":
             return cls.w_mxfp6_e3m2_a_mxfp6_e3m2
-        elif input_dtype == "fp8" and weight_dtype == "mxfp6_e3m2":
-            return cls.w_mxfp6_e3m2_a_fp8
         elif input_dtype == "mxfp6_e2m3" and weight_dtype == "mxfp6_e2m3":
             return cls.w_mxfp6_e2m3_a_mxfp6_e2m3
-        elif input_dtype == "fp8" and weight_dtype == "mxfp6_e2m3":
-            return cls.w_mxfp6_e2m3_a_fp8
         else:
             logger.warning(
                 "input_dtype='%s' and"
