--- a/vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py
+++ b/vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py
@@ -1,16 +1,14 @@
 # SPDX-License-Identifier: Apache-2.0
 # SPDX-FileCopyrightText: Copyright contributors to the vLLM project
 
-from collections.abc import Callable
 from fractions import Fraction
 from functools import cache, partial
-from typing import Any
+from typing import Any, Callable, Optional, Union
 
 import torch
 import torch.nn.functional as F
 
 from vllm import envs
-from vllm._aiter_ops import rocm_aiter_ops
 from vllm.logger import init_logger
 from vllm.model_executor.layers.quantization.utils.mxfp4_utils import (
     dequant_mxfp4,
@@ -24,12 +22,7 @@
     OCP_MX_BLOCK_SIZE,
     OCP_MX_Scheme,
 )
-from vllm.model_executor.parameter import (
-    GroupQuantScaleParameter,
-    ModelWeightParameter,
-    PackedvLLMParameter,
-)
-from vllm.model_executor.utils import set_weight_attrs
+from vllm.model_executor.parameter import GroupQuantScaleParameter, PackedvLLMParameter
 from vllm.platforms import current_platform
 
 from .quark_scheme import QuarkScheme
@@ -37,13 +30,6 @@
 logger = init_logger(__name__)
 
 
-# TODO: move registration of custom op to aiter_ops.py
-# `from vllm._aiter_ops import rocm_aiter_ops`
-# use `rocm_aiter_ops.is_asm_fp4_gemm_dynamic_quant_enabled()`
-# for envs checks which does not require @cache anymore.
-# triton kernel is torch compile compatible.
-# does not require direct registration.
-# use `rocm_aiter_ops.triton_fp4_gemm_dynamic_qaunt`.
 @cache
 def is_rocm_aiter_fp4_asm_gemm_enabled() -> bool:
     return (
@@ -55,13 +41,10 @@
 
 try:
     from aiter.ops.shuffle import shuffle_weight
-    from aiter.ops.triton.gemm_afp4wfp4 import (
-        gemm_afp4wfp4,
-        gemm_afp4wfp4_preshuffled_weight_scales,
-    )
+    from aiter.ops.triton.gemm_afp4wfp4 import gemm_afp4wfp4
     from aiter.ops.triton.quant import dynamic_mxfp4_quant
 
-    from vllm.utils.torch_utils import direct_register_custom_op
+    from vllm.utils import direct_register_custom_op
 
     if is_rocm_aiter_fp4_asm_gemm_enabled():
         from aiter import gemm_a4w4, per_1x32_f4_quant_hip
@@ -71,65 +54,27 @@
         weight: torch.Tensor,
         weight_scale: torch.Tensor,
         rocm_use_aiter_fp4_asm_gemm: bool = False,
-        out_dtype: torch.dtype | None = torch.bfloat16,
-        x_scales: torch.Tensor | None = None,
+        out_dtype: Optional[torch.dtype] = torch.bfloat16,
+        x_scales: Optional[torch.Tensor] = None,
     ) -> torch.Tensor:
         M = x.shape[0]
-        N = weight.shape[0]
-        K = weight.shape[1]
         if rocm_use_aiter_fp4_asm_gemm:
-            if M <= 64 and rocm_aiter_ops.is_triton_gemm_afp4wfp4_presh_ws_tuned(N, K):
-                if x_scales is None:
-                    # use hip quant kernel for performance
-                    if M >= 32:
-                        x_q, x_s = per_1x32_f4_quant_hip(x, shuffle=True)
-                    else:
-                        x_q, x_s = per_1x32_f4_quant_hip(x, shuffle=False)
-                else:
-                    x_q = x
-                    x_s = x_scales
-
-                if M >= 32:
-                    x_s = x_s.view(torch.uint8).view(x_s.shape[0] // 32, -1)
-                else:
-                    x_s = x_s[:M, ...].view(torch.uint8)
-
-                y = torch.empty(M, N, device=x_q.device, dtype=out_dtype)
-                gemm_afp4wfp4_preshuffled_weight_scales(
-                    x_q.view(torch.uint8),
-                    weight.view(torch.uint8).view(weight.shape[0] // 16, -1),
-                    x_s,
-                    weight_scale.view(torch.uint8).view(
-                        weight_scale.shape[0] // 32, -1
-                    ),
-                    out_dtype,
-                    y,
-                )
+            if x_scales is None:
+                # use hip quant kernel for performance
+                x_q, x_s = per_1x32_f4_quant_hip(x, shuffle=True)
             else:
-                if x_scales is None:
-                    # use hip quant kernel for performance
-                    x_q, x_s = per_1x32_f4_quant_hip(x, shuffle=True)
-                else:
-                    x_q = x
-                    x_s = x_scales
-
-                # 32 alignment is enough for dim0 padding of output for
-                # gemm_a4w4 kernel
-                y = torch.empty(
-                    (M + 31) // 32 * 32,
-                    weight.shape[0],
-                    device=x_q.device,
-                    dtype=out_dtype,
-                )
-
-                gemm_a4w4(
-                    x_q,
-                    weight.view(x_q.dtype),
-                    x_s,
-                    weight_scale.view(x_s.dtype),
-                    y,
-                    bpreshuffle=True,
-                )
+                x_q = x
+                x_s = x_scales
+
+            # 32 alignment is enough for dim0 padding of output for
+            # gemm_a4w4 kernel
+            y = torch.empty(
+                (M + 31) // 32 * 32, weight.shape[0], device=x_q.device, dtype=out_dtype
+            )
+
+            gemm_a4w4(
+                x_q, weight, x_s, weight_scale.view(x_s.dtype), y, bpreshuffle=True
+            )
             return y[:M]
         else:
             if x_scales is None:
@@ -150,7 +95,7 @@
         weight_scale: torch.Tensor,
         x_scales: torch.Tensor = None,
         rocm_use_aiter_fp4_asm_gemm: bool = False,
-        out_dtype: torch.dtype | None = torch.bfloat16,
+        out_dtype: Optional[torch.dtype] = torch.bfloat16,
     ) -> torch.Tensor:
         return torch.empty(
             (*x.shape[:-1], weight.shape[0]), dtype=out_dtype, device=x.device
@@ -163,27 +108,19 @@
         fake_impl=gemm_with_dynamic_quant_fake,
         dispatch_key=current_platform.dispatch_key,
     )
-except (ImportError, AttributeError, RuntimeError):
-    if current_platform.is_rocm():
-        logger.warning(
-            "AITER is not found or QuarkOCP_MX is not supported on the current "
-            "platform. QuarkOCP_MX quantization will not be available."
-        )
+except (ImportError, AttributeError):
     dynamic_mxfp4_quant = gemm_afp4wfp4 = None
 
 
 class QuarkOCP_MX(QuarkScheme):
     def __init__(
-        self,
-        weight_quant_spec: dict[str, Any],
-        input_quant_spec: dict[str, Any],
-        dynamic_mxfp4_quant: bool = False,
+        self, weight_quant_spec: dict[str, Any], input_quant_spec: dict[str, Any]
     ):
         self.out_dtype = torch.get_default_dtype()
         self.qscheme = "per_group"
         self.weight_quant_spec = weight_quant_spec
         self.input_quant_spec = input_quant_spec
-        self.dynamic_mxfp4_quant = dynamic_mxfp4_quant
+
         self.weight_dtype = weight_quant_spec["dtype"].replace("fp", "mxfp")
         self.input_dtype = input_quant_spec["dtype"].replace("fp", "mxfp")
 
@@ -192,7 +129,7 @@
         )
 
         if self.weight_dtype == "mxfp4":
-            self.packed_factor: int | Fraction = 2
+            self.packed_factor: Union[int, Fraction] = 2
             self.dequant_func = dequant_mxfp4
         else:
             self.packed_factor = Fraction(numerator=8, denominator=6)
@@ -277,13 +214,7 @@
                 layer.weight_scale.data, requires_grad=False
             )
         else:
-            if self.dynamic_mxfp4_quant:
-                w_q, w_s = dynamic_mxfp4_quant(layer.weight)
-                layer.weight_scale = torch.nn.Parameter(
-                    w_s.T.contiguous(), requires_grad=False
-                )
-                layer.weight = torch.nn.Parameter(w_q, requires_grad=False)
-            elif self.rocm_use_aiter_fp4_asm_gemm:
+            if self.rocm_use_aiter_fp4_asm_gemm:
                 # shuffle weight scale
                 weight_scale_shuffle = layer.weight_scale.data
                 sm, sn = weight_scale_shuffle.shape
@@ -316,57 +247,42 @@
         weight_loader: Callable,
         **kwargs,
     ):
-        if self.dynamic_mxfp4_quant:
-            weight = ModelWeightParameter(
-                data=torch.empty(
-                    sum(output_partition_sizes),
-                    input_size_per_partition,
-                    dtype=params_dtype,
-                ),
-                input_dim=1,
-                output_dim=0,
-                weight_loader=weight_loader,
-            )
-
-            layer.register_parameter("weight", weight)
-            set_weight_attrs(weight, kwargs)
-        else:
-            output_size_per_partition = sum(output_partition_sizes)
-            layer.logical_widths = output_partition_sizes
-
-            # WEIGHT
-            weight = PackedvLLMParameter(
-                data=torch.empty(
-                    output_size_per_partition,
-                    self.get_packed_dim(input_size_per_partition, self.weight_dtype),
-                    dtype=torch.uint8,
-                ),
-                input_dim=1,
-                output_dim=0,
-                packed_dim=1,
-                packed_factor=self.packed_factor,
-                weight_loader=weight_loader,
-            )
-            layer.register_parameter("weight", weight)
-
-            # WEIGHT SCALE
-            weight_scale = GroupQuantScaleParameter(
-                data=torch.empty(
-                    output_size_per_partition,
-                    input_size_per_partition // OCP_MX_BLOCK_SIZE,
-                    dtype=torch.uint8,
-                ),
-                input_dim=1,
-                output_dim=0,
-                weight_loader=weight_loader,
-            )
-            layer.register_parameter("weight_scale", weight_scale)
+        output_size_per_partition = sum(output_partition_sizes)
+        layer.logical_widths = output_partition_sizes
+
+        # WEIGHT
+        weight = PackedvLLMParameter(
+            data=torch.empty(
+                output_size_per_partition,
+                self.get_packed_dim(input_size_per_partition, self.weight_dtype),
+                dtype=torch.uint8,
+            ),
+            input_dim=1,
+            output_dim=0,
+            packed_dim=1,
+            packed_factor=self.packed_factor,
+            weight_loader=weight_loader,
+        )
+        layer.register_parameter("weight", weight)
+
+        # WEIGHT SCALE
+        weight_scale = GroupQuantScaleParameter(
+            data=torch.empty(
+                output_size_per_partition,
+                input_size_per_partition // OCP_MX_BLOCK_SIZE,
+                dtype=torch.uint8,
+            ),
+            input_dim=1,
+            output_dim=0,
+            weight_loader=weight_loader,
+        )
+        layer.register_parameter("weight_scale", weight_scale)
 
     def apply_weights(
         self,
         layer: torch.nn.Module,
         x: torch.Tensor,
-        bias: torch.Tensor | None = None,
+        bias: Optional[torch.Tensor] = None,
     ) -> torch.Tensor:
         if self.emulate:
             dq_w = self.dequant_func(layer.weight, layer.weight_scale, x.dtype)
