--- a/vllm/transformers_utils/config.py
+++ b/vllm/transformers_utils/config.py
@@ -79,7 +79,6 @@
     Llama_Nemotron_Nano_VL="Nemotron_Nano_VL_Config",
     RefinedWeb="RWConfig",  # For tiiuae/falcon-40b(-instruct)
     RefinedWebModel="RWConfig",  # For tiiuae/falcon-7b(-instruct)
-    fastconformer_ctc="FastConformerCTCConfig",
     jais="JAISConfig",
     mlp_speculator="MLPSpeculatorConfig",
     medusa="MedusaConfig",
